app/src/main/java/edu/uc/groupProject/topten/service/FirestoreService.kt

```diff
@@ -131,7 +131,7 @@ class FirestoreService {
                 if (document != null) {
                     val allListItems = ArrayList<ListItem>()
 
-                    var listItem: ListItem = ListItem(
+                    val listItem: ListItem = ListItem(
```

Comment:
It is practice is to use val until you decide you need to modify it (in which case you go back and define it as a var). "**Favoring Immutability**" is the principle here. There is always the chance a variable value will change unexpectedly, so this way you can be signaled when that is happening and can then decide if it's good functionality (and change it to a var) or if something is off. Here is a website that talks some about it along with other Kotlin Best Practices: https://batchofcode.com/2018/04/13/kotlin-best-practices.html#1-embrace-immutability

Reply:
Good idea.